Sources/tart/Credentials/DockerConfigCredentialsProvider.swift

```diff
@@ -41,13 +41,17 @@ class DockerConfigCredentialsProvider: CredentialsProvider {
 
     process.waitUntilExit()
 
+    let outputData = try outPipe.fileHandleForReading.readToEnd()
     if !(process.terminationReason == .exit && process.terminationStatus == 0) {
+      let outputText = outputData == nil ? "" : String(decoding: outputData!, as: UTF8.self)
+      print(outputText)
       throw CredentialsProviderError.Failed(message: "Docker helper failed!")
     }
+    if outputData == nil {
```

Comment:
Could the `outputData` be non-nil, but still empty?

Reply:
Might be but it will mean that the helper binary breaks contract by not outputting anything. JSON decoding will break then.

Reply:
But can we check for the `outputData?.count == 0` instead?

That will be much cleaner IMO 🤔

Reply:
As you wish! See [a637f10](https://github.com/cirruslabs/tart/commit/a637f1093f6698dbcba0844fae02b7c15cd7c23a) [now](https://github.com/cirruslabs/tart/pull/654#event-10928229153)